### PR TITLE
<IRIP-43> refactor models/schemas

### PR DESCRIPTION
**Pull request by rG480**
Refactoring from typeorm to sequelize is done, with some initial tests written as well.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,24 +1,47 @@
...
-import express from 'express'
+//import "reflect-metadata";
 import { ApolloServer } from 'apollo-server-express';
+import express from 'express';
+import sequelize from 'sequelize'
+import {Sequelize} from 'sequelize-typescript'
 import { buildSchema } from "type-graphql";
-
+import {Chart} from './models/Chart'
+import {Song} from  './models/Song'
```
Comment (Reyder95) [inline src/index.ts:7]: At some point down the line (either in a while or relatively soon), we should make an `index.ts` file that takes these and exports them through one file (so we will only need one import).

File: src/index.ts [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
+import {User} from './models/User'
+import {Platform} from './models/Platform'
 import UserResolver from './resolvers/UserResolver';
 import PlatformResolver from './resolvers/PlatformResolver';
 import SongResolver from './resolvers/SongResolver';
 import ChartResolver from "./resolvers/ChartResolver";
```
Comment (Reyder95) [inline src/index.ts:13]: Same thing for these? Making an `index.ts` so we can use one import.

File: src/models/Chart.ts [hunk 3]
```diff
@@ -1,47 +1,55 @@
...
+import { Column, Model, BelongsTo, Table, ForeignKey } from 'sequelize-typescript';
 import { ObjectType, Field, ID } from "type-graphql";
+import {Song} from './Song'
 
 
 //Chart(s) model
-@Entity()
+@Table
 @ObjectType()
-export class Chart extends BaseEntity{
```
Comment (Reyder95) [inline src/models/Chart.ts:8]: If you do 

```
/**
 * some stuff
 */
```

Typescript will put that into its intellisense whenever you use the function, entity, or class that you put this above. Maybe we should start doing that.

File: src/models/Chart.ts [hunk 4]
```diff
@@ -1,47 +1,55 @@
...
-    
+export class Chart extends Model{
+    //id primary key column generated implicity by sequelize
+
     //ID number that maps a particular chart to it's corresponding song
+    @BelongsTo(() => Song)
+    song:Song;
     @Field(()=>Number)
-    @Column({name:'song_id'})
-    songId:number;
```
Comment (Reyder95) [inline src/models/Chart.ts:16]: We should probably put NOT NULL on a lot of these fields, honestly.

File: src/models/Platform.ts [hunk 5]
```diff
@@ -24,6 +22,12 @@ export class Platform extends BaseEntity {
    *  between different releases of the client.
    */
   @Field(() => String)
-  @Column()
+  @Column
   versionCode: string;
```
Comment (Reyder95) [inline src/models/Platform.ts:26]: Limit this to a number of characters like 5. The current version codes for sims that exist are KSM and USC. And probably DATA if we want to go there. I doubt any other version code will go beyond that, and if they do this is a quick update. Just want to keep things consistent and set-in-stone.

File: src/models/User.ts [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
-  @Field(() => ID) // ID Field
-  @PrimaryGeneratedColumn() // Primary key
-  id: number;
+export class User extends Model {
+ //id primary key column generated implicity by sequelize
 
   @Field(() => String) // String field
-  @Column() // Basic column
+  @Column// Basic column
   username: string;
```
Comment (Reyder95) [inline src/models/User.ts:14]: Should be unique

File: src/resolvers/SongResolver.ts [hunk 7]
```diff
@@ -1,40 +1,57 @@
...
+import{Chart} from '../models/Chart';
 import { CreateSongInput, UpdateSongInput } from '../inputs/SongInput';
+import { ApolloError } from 'apollo-server-errors';
 
 @Resolver()
 class SongResolver {
-  /** Find all songs currently in the database. */
+  // Find all songs currently in the database. 
   @Query(() => [Song])
   songs() {
```
Comment (Reyder95) [inline src/resolvers/SongResolver.ts:11]: I want to change our naming convention for this I think... but idk.

What do you think about making these getEntity and getSingleEntity. Or maybe just singleEntity and entities.

Having them differ by one letter might be confusing...

File: src/index.ts (hunk 1 above)
Comment (rG480) [reply src/index.ts:7]: I agree; I'll definitely get to work on consolidating the models/resolvers into a single index file present in both folders.

File: src/index.ts (hunk 2 above)
Comment (rG480) [reply src/index.ts:13]: I'll work on something to do this in the coming days for both models and resolvers.

File: src/resolvers/SongResolver.ts (hunk 7 above)
Comment (rG480) [reply src/resolvers/SongResolver.ts:11]: +1 for the name change to reduce ambiguity. Something like getAllSongs/ getSong might work but let's see what everyone thinks. It's a simple change.

**Pull request merged**